addons/mail/models/mail_channel.py

```diff
@@ -318,6 +318,8 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                         group_name=channel.group_public_id.name,
                         partner_names=', '.join(partner.name for partner in invalid_partners)
                     ))
+                if guests:
+                    raise UserError(_("Not Found"))
```

Comment:
You may want to describe in a bit more detail what went wrong in the error message, for example "Cannot add guests to groups-restricted channels".

Reply:
Ideally with the channel and group names on it, similar to the error above.

And don't forget to export the translation file afterwards.